### PR TITLE
CONTRIBUTING.md has a new code standard about naming arguments passed to vars

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request & Why It's Good For The Game

Title. There has been some really hideous arg names for this situation of passing it into a var, for example:

`/datum/element/decal/Attach(atom/target, _icon, _icon_state, _dir, _cleanable=FALSE, _color, _layer=TURF_LAYER, _description, _alpha=255, mutable_appearance/_pic)`

Some of which aren't even matching the format of var = _arg...

image

## Changelog
Not player facing

### CODE REVIEW

**Comment by TemporalOroboros** [comment]
Honestly glad to have this cleared up.

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -643,6 +643,50 @@ Proc variables, static variables, and global variables are resolved at compile t
...
 
+### When passing vars through New() or Initialize()'s arguments, use src.var
+Using src.var + naming the arguments the same as the var is the most readable and intuitive way to pass arguments into a new instance's vars. The main benefit is that you do not need to give arguments odd names with prefixes and suffixes that are easily forgotten in `new()` when sending named args.
+
+This is very bad:
+```DM
+/atom/thing
+	var/is_red
+
+var/atom/thing/Initialize(mapload, enable_red)
```
Comment (Mothblocks) [inline .github/CONTRIBUTING.md:654]:
```suggestion
/atom/thing/Initialize(mapload, enable_red)
```

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -643,6 +643,50 @@ Proc variables, static variables, and global variables are resolved at compile t
...
+```
+
+Future coders using this code will have to remember two differently named variables which are near-synonyms of eachother. One of them is only used in Initialize for one line.
+
+This is bad:
+```DM
+/atom/thing
+	var/is_red
+
+var/atom/thing/Initialize(mapload, _is_red)
```
Comment (Mothblocks) [inline .github/CONTRIBUTING.md:668]:
```suggestion
/atom/thing/Initialize(mapload, _is_red)
```

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -643,6 +643,50 @@ Proc variables, static variables, and global variables are resolved at compile t
...
+```
+
+`_is_red` is being used to set `is_red` and yet means a random '_' needs to be appended to the front of the arg, same as all other args like this.
+
+This is good:
+```DM
+/atom/thing
+	var/is_red
+
+var/atom/thing/Initialize(mapload, is_red)
```
Comment (Mothblocks) [inline .github/CONTRIBUTING.md:682]:
```suggestion
/atom/thing/Initialize(mapload, is_red)
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by LemonInTheDark** [comment]
if I'm honest I don't really like `src.varname = varname`, it seems like a bit of a mind flood. I do hate `varname =_varname` even more, but I'm not sure if we should insitute src.var as a rule.

Is there maintainer support for this? @Mothblocks you want this right?

**Comment by Mothblocks** [comment]
Yes. 🙂

**Comment by tralezab** [comment]
The alternates to _var are actually way worse, since the argument passed as a variable will only be used in a single line, yet require naming it as a synonym of the var set

**Comment by Mothblocks** [comment]
CC @tgstation/commit-access 

This is something I've been enforcing for a while and it makes a lot of things, most often component code, a lot better.

**Comment by tralezab** [comment]
everywhere i look

i find more cases of BOTH making it a synonym AND adding an underscore

image

**Comment by optimumtact** [comment]
I refuse to use such ugly underscores, when src.var = var works perfectly fine

**Comment by Cyberboss** [comment]
I hate named arguments to begin with and would rather see them all eliminated.

Since that's not going to happen though, I guess this is the next best thing.

**Pull request merged**